metrics-registry.yaml

```diff
@@ -0,0 +1,240 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
```

Comment:
What do you think about using sections to organize the registry?  I think YAML calls them nested levels.  For example:

```
asset:
    - name: updates
      type: counter
      description: ""
dag_processing:
    - name: manager_stalls
      type: counter
      description: ""
    - name: other_callback_count
      type: counter
      description: ""
... etc...
```

Maybe experiment with that and with what the code to process it would look like before we commit to a flat structure like this?  We have a bunch of `.yml` files int he codebase you can look at for inspiration and to see how they are parsed in code.

Reply:
I agree it looks neater that way but my rational for the flat structure is that the [metrics docs](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/metrics.html#counters) is a flat list, so we save some parsing/de-parsing by having a flat structure. 
Let me know what you think

Reply:
Also if we sort it alphabetically, the registry will be naturally "organized" into the respective groups as well